src/stan/variational/advi.hpp

```diff
@@ -334,6 +357,14 @@ namespace stan {
 
         // draw more samples from posterior and write on subsequent lines
         if (out_stream_) {
+          if (print_stream_) {
```

Comment:
Can print_stream_ exist if out_stream_ doesn't exist?

Also, you almost never want to do

```
if (cond1) { if (cond2) { ... }}
```

It's better in terms of efficiency and readability as:

```
if (cond1 && cond2) { ... }
```

- Bob

P.S.  I really do not like these 0 pointer patterns.
The are dangerous because they open up easy segfault
opportunities because <b>every single use</b> of `out_stream_`
with `if (out_stream_)`.

> On Nov 3, 2015, at 9:48 AM, Dustin Tran <notifications@github.com> wrote:
> 
> In src/stan/variational/advi.hpp:
> 
> > @@ -334,6 +357,14 @@ namespace stan {
> >  
> >          // draw more samples from posterior and write on subsequent lines
> >          if (out_stream_) {
> > +          if (print_stream_) {
> 
> if out_stream_ doesn't exist, then the algorithm doesn't need to draw any samples. (that's currently how it works)
> 
> —
> Reply to this email directly or view it on GitHub.
>